### PR TITLE
BibAuthorID: ORCID support for rabbit

### PR DESCRIPTION
**Pull request by atsikiridis**
- Rabbit tries to match papers to profiles
  by ORCID, after trying to do so with an
  INSPIRE id.
- The ORCID is located in $__j fields with
  the prefix of 'ORCID:'.

Signed-off-by: Artem Tsikiridis artem.tsikiridis@cern.ch

### CODE REVIEW

File: modules/bibauthorid/lib/bibauthorid_dbinterface.py [hunk 1]
```diff
@@ -4322,17 +4322,24 @@ def get_inspire_id_of_signature(sig):  # get_inspire_id
...
+    @rtype: str
+    """
+    table, ref, rec = sig
+    ext_ids_field = (str(table), ref, rec), str(table) + '__j'
 
-    return None
+    ext_ids = get_grouped_records(*ext_ids_field).values()[0]
+    for ext_id in ext_ids:
+        if ext_id.startswith('ORCID:'):
+            return ext_id
```
Comment (glouppe) [inline modules/bibauthorid/lib/bibauthorid_dbinterface.py]: We should in fact extract the ORCID number of `ext_id` to match the storage format in aidPERSONIDDATA. That is return `m.group()` if `m = re.search(r"(\d{4}\-){3}\d{4}", ext_id)` is not None.

**Comment by glouppe** [comment]
+1 for merge once my comment is addressed.

File: modules/bibauthorid/lib/bibauthorid_dbinterface.py (hunk 1 above)
Comment (kaplun) [inline modules/bibauthorid/lib/bibauthorid_dbinterface.py]: +1

**Comment by atsikiridis** [comment]
@glouppe @kaplun  I think now it is ok. I also added an additional regression test for rabbit.

**Comment by glouppe** [comment]
Changes look fine with me now. +1 for merge.

**Comment by kaplun** [comment]
Since I don't see Drone has run on this last branch, do you confirm unit/regression tests are passing on this branch?

**Comment by atsikiridis** [comment]
@kaplun Yes, I can confirm that the tests are passing on my machine. I amended something so let's see what drone says now.

**Pull request merged**